File: frontend/src/utils/privacy.js
import { marked } from 'marked'

marked.setOptions({ breaks: true, gfm: false })

export function maskPhone(phone) {
  if (!phone || phone.length !== 11) return phone
  return phone.slice(0, 3) + '****' + phone.slice(7)
}

export function maskPhoneInText(text) {
  if (!text) return text
  return text.replace(/(1[3-9]\d)\d{4}(\d{4})/g, '$1****$2')
}

export function renderContent(text, role) {
  if (!text) return ''
  let result = maskPhoneInText(text)
  if (role === 'assistant') {
    result = marked.parse(result)
  }
  return result
}

File: backend/app/utils/privacy.py
import re


def mask_phone(text: str) -> str:
    """将文本中的手机号替换为脱敏格式（138****1234）"""
    if not text:
        return text
    return re.sub(r"(1[3-9]\d)\d{4}(\d{4})", r"\1****\2", text)


def mask_dict_phone(data: dict, key: str = "phone") -> dict:
    """对字典中指定 key 的手机号字段脱敏，返回新字典"""
    if not data:
        return data
    result = dict(data)
    if key in result and isinstance(result[key], str):
        result[key] = mask_phone(result[key])
    return result
